openmls/src/framing/public_message.rs

```diff
@@ -196,47 +218,24 @@ impl PublicMessage {
 
     /// Get the group epoch.
     pub(crate) fn epoch(&self) -> GroupEpoch {
-        self.content.epoch
+        self.content.content.epoch
     }
 
     /// Get the [`GroupId`].
     pub(crate) fn group_id(&self) -> &GroupId {
-        &self.content.group_id
-    }
-
-    fn encode_tbs<'a>(
-        &self,
-        serialized_context: impl Into<Option<&'a [u8]>>,
-    ) -> Result<Vec<u8>, tls_codec::Error> {
-        let mut out = Vec::new();
-        FramedContentTbs::serialize_content_tbs(
-            WireFormat::PublicMessage,
-            &self.content,
-            serialized_context,
-            &mut out,
-        )?;
-        Ok(out)
+        &self.content.content.group_id
     }
 
-    /// Turns this [`PublicMessage`] into a [`VerifiableAuthenticatedContent`]. The
-    /// `serialized_context` is added only if the [`Sender`] of the plaintext is
-    /// [`Sender::Member`] or [`Sender::NewMemberCommit`] as per MLS specification.
-    pub fn into_verifiable_content(
-        self,
-        serialized_context: &[u8],
-    ) -> VerifiableAuthenticatedContent {
+    /// Set the context for later verification. This should only be done for
+    /// messages with [`Sender::Member`] and [`Sender::NewMemberCommit`].
+    pub(super) fn set_context(&mut self, context: &[u8]) {
```

Comment:
It shouldn't only be done for these but it won't happen for any other sender.
You call it unconditionally and the function sets it correctly. Maybe change the comment to say that? What it currently says is a little confusing.

Reply:
Good point, I'll change the comment. Again, we probably want to enforce this on a type level at some point, but maybe not in this PR.